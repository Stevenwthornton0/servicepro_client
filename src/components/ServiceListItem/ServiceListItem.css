.serviceLink {
    width: 100%;
    margin-bottom: 26px;
    margin-left: auto;
    margin-right: auto;
    padding: 0 30px 0 30px;
}

.serviceLink > h3 {
    font-size: 22px;
}

.serviceItem {
    border: 1px solid black;
    border-radius: 4px;
    width: 80%;
    margin-left: auto;
    margin-right: auto;
    margin-bottom: 20px;
}

.serviceItem:hover {
    background-color: lightblue;
}

.serviceItem > a {
    text-decoration: none;
    color: black;
}

.serviceDelete {
    width: 30%;
    height: 30px;
    margin-left: 30px;
    margin-bottom: 15px;
    border: 1px solid #bdbdbd;
    border-radius: 4px;
    background-color: lightblue;
}

.bold {
    font-weight: 700;
}

.notBold {
    font-weight: 500;
}

@media screen and (min-width: 770px) {
    .serviceItem {
        width: 900px;
        margin-bottom: 40px;
    }

    .serviceDelete {
        width: 15%;
    }

    .serviceDelete:hover {
        background-color: #424242;
        border: 1px solid #424242;
        color: lightblue;
        cursor: pointer;
    }
}