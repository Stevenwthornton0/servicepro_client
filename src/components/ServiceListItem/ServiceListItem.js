import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import ServicesContext from '../../contexts/ServicesContext';
import { displayPhone } from '../../Utils/Utils';
import './ServiceListItem.css';

class ServiceListItem extends Component {
    static contextType = ServicesContext;

    static defaultProps = {
        service: [],
        admin: false,
        deleteService: () => {}
    }

    truncate = (input) => {
        if (input.length > 150)
           return input.substring(0, 150) + '...';
        else
           return input;
     };

    renderDeleteButton(admin, serviceId) {
        const { deleteService } = this.props;
        if (admin) {
            return ( <button className='serviceDelete' onClick={e => deleteService(serviceId)}>delete</button>) 
        }

        return null
    }

    render() {
        const { service, admin } = this.props;
        let { city, state } = service;
        if (city && state) {
            city = city.replace(/^\w/, c => c.toUpperCase());
            state = state.replace(/^\w/, c => c.toUpperCase());
        }
        return (
            <div className='serviceItem'>
                <Link to={`/services/service/${service.id}`}>
                    <section className='serviceLink'>
                        <h3>{service.name}</h3>
                        <p className='bold'>{city}, {state}</p>
                        <p>{this.truncate(service.about)}</p>
                        <p className='bold'>Email: <span className='notBold'>{service.email}</span></p>
                        <p className='bold'>Phone: {displayPhone(service)}</p>
                    </section>
                </Link>
                {this.renderDeleteButton(admin, service.id)}
            </div>
        )
    }
}

export default ServiceListItem;