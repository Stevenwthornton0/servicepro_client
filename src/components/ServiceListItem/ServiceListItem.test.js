import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme'
import ServiceListItem from './ServiceListItem';


Enzyme.configure({ adapter: new Adapter() })

describe('ServiceListItem component', () => {
    it('renders without crashing', () => {
        const service = {
            id: 17,
            service_type: "mechanic",
            name: "Test Mechanic",
            email: "test@mchanic.com",
            phone: "1234561234",
            about: "Test"
        }
        const wrapper = shallow(<ServiceListItem service={service}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})