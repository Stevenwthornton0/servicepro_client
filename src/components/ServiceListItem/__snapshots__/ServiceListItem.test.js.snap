// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceListItem component renders without crashing 1`] = `
<div
  className="serviceItem"
>
  <Link
    to="/services/service/17"
  >
    <section
      className="serviceLink"
    >
      <h3>
        Test Mechanic
      </h3>
      <p>
        Test
      </p>
      <p
        className="bold"
      >
        Email: 
        <span
          className="notBold"
        >
          test@mchanic.com
        </span>
      </p>
      <p
        className="bold"
      >
        Phone: 
        <span
          className="notBold phone"
        >
          123-456-1234
        </span>
      </p>
    </section>
  </Link>
</div>
`;
